app/scripts/components/common/aoi/use-aoi-controls.ts

```diff
@@ -16,68 +22,130 @@ export function useAoiControls(initialState: Partial<AoiState> = {}) {
     setAoi({
       drawing: false,
       selected: false,
-      feature: null,
+      selectedContext: undefined,
+      featureCollection: null,
       actionOrigin: null,
       ...initialState
     });
   }, [initialState]);
 
   const onAoiEvent = useCallback((action, payload) => {
     switch (action) {
+      case 'aoi.trash-click': {
+        // We need to programmatically access the mapbox draw trash method which
+        // will do different things depending on the selected mode.
+        // @ts-expect-error _drawControl does exist but it is an internal
+        // property.
+        const mbDraw = mapRef.current?.instance?._drawControl;
+        if (!mbDraw) return;
+        mbDraw.trash();
+        break;
+      }
       case 'aoi.draw-click':
-        // There can only be one selection (feature) on the map
-        // If there's a feature toggle the selection.
-        // If there's no feature toggle the drawing.
         setAoi((state) => {
-          const selected = !!state.feature && !state.selected;
           return {
             ...state,
-            drawing: !state.feature && !state.drawing,
-            selected,
-            actionOrigin: selected ? 'panel' : null
+            drawing: true,
+            selected: false,
+            selectedContext: undefined,
+            actionOrigin: null
+          };
+        });
+        break;
+      case 'aoi.select-click':
+        setAoi((state) => {
+          return {
+            ...state,
+            drawing: false,
+            selected: false,
+            selectedContext: undefined,
+            actionOrigin: null
```

Comment:
Throughout that switch statement there's a lot of resetting state going on, could we DRY it a little bit?

```suggestion
            ...DEFAULT_PARAMETERS
```

Reply:
![](https://i.giphy.com/media/eZ2fk4Z0bKWQ0/giphy-downsized.gif)